.github/workflows/update-crates-versions.yml

```diff
@@ -0,0 +1,38 @@
+name: Update crates versions
+
+defaults:
+  run:
+    shell: bash
+
+on:
+  schedule:
+    # Every Sunday at 00:00
+    - cron:  '0 0 * * 0'
+  workflow_dispatch:
+
+jobs:
+  updateversions:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the code
+        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579
+        with:
+          fetch-depth: 1
+      - name: Setup rust toolchain
+        uses: actions-rs/toolchain@16499b5e05bf2e26879000db0c1d13f7e13fa3af
+        with:
+          toolchain: stable
+          default: true
+      - name: Retrieve newest versions of dependencies
+        run: ./bin/update-crates-versions.sh
+      - name: Create pull request
```

Comment:
I would maybe split this one step into several steps, like https://github.com/exercism/org-wide-files/blob/main/.github/workflows/sync-rest.yml#L303 That makes it easier to debug what went wrong when a build fails.